narwhals/_pandas_like/namespace.py

```diff
@@ -250,3 +250,116 @@ def concat(
                 backend_version=self._backend_version,
             )
         raise NotImplementedError
+
+    def when(
+        self,
+        *predicates: IntoPandasLikeExpr,
+    ) -> PandasWhen:
+        plx = self.__class__(self._implementation, self._backend_version)
+        if predicates:
+            condition = plx.all_horizontal(*predicates)
+        else:
+            msg = "at least one predicate needs to be provided"
+            raise TypeError(msg)
+
+        return PandasWhen(condition, self._implementation, self._backend_version)
+
+
+def _when_then_value_arg_process(
+    plx: PandasLikeNamespace,
+    value: PandasLikeExpr | PandasLikeSeries | Any,
+    *,
+    shape: tuple[int],
+) -> PandasLikeExpr:
+    from narwhals.dependencies import get_numpy
+
+    # NumPy is a required dependency of pandas
+    np = get_numpy()
+
+    if isinstance(value, PandasLikeExpr):
+        return value
+    elif isinstance(value, PandasLikeSeries):
+        return plx._create_expr_from_series(value)
+    elif (np := get_numpy()) is not None and isinstance(value, np.ndarray):
```

Comment:
As numpy is already available from line 277, there is no need for it here:

```suggestion
    elif isinstance(value, np.ndarray):
```